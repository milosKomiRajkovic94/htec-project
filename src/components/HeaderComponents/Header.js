import React from "react";
import PropTypes from "prop-types";

import { useTranslation } from "react-i18next";

import { connect } from "react-redux";

import { TOP_NEWS, CATEGORIES, SEARCH } from "../../constants/urls";

import { LANGUAGE_EN, LANGUAGE_US } from "../../constants/stringConstants";

const Header = ({
  changeLocation,
  changeLanguage,
  currentLocation,
  currentLanguage,
}) => {
  const { t } = useTranslation();

  return (
    <header className="header">
      <div className="headerThreeLeftOptions">
        <div
          onClick={() => changeLocation(TOP_NEWS)}
          className={
            currentLocation === TOP_NEWS
              ? "headerOption headerOptionHighlighted"
              : "headerOption"
          }
          role="button"
          tabIndex={0}
          onKeyPress={() => changeLocation(TOP_NEWS)}
        >
          {t("topNews")}
        </div>
        <div
          onClick={() => changeLocation(CATEGORIES)}
          className={
            currentLocation === CATEGORIES
              ? "headerOption headerOptionHighlighted"
              : "headerOption"
          }
          role="button"
          tabIndex={0}
          onKeyPress={() => changeLocation(CATEGORIES)}
        >
          {t("categories")}
        </div>
        <div
          onClick={() => changeLocation(SEARCH)}
          className={
            currentLocation === SEARCH
              ? "headerOption headerOptionHighlighted"
              : "headerOption"
          }
          role="button"
          tabIndex={0}
          onKeyPress={() => changeLocation(SEARCH)}
        >
          {t("search")}
        </div>
      </div>
      <div className="headerTwoRightOptions">
        <div
          onClick={() => changeLanguage(LANGUAGE_EN)}
          className={
            currentLanguage === LANGUAGE_EN
              ? "headerOption headerOptionHighlighted"
              : "headerOption"
          }
          role="button"
          id="gbLanguage"
          data-testid="gbLanguage"
          tabIndex={0}
          onKeyPress={() => changeLanguage(LANGUAGE_EN)}
        >
          {t("gb")}
        </div>
        <div
          onClick={() => changeLanguage(LANGUAGE_US)}
          className={
            currentLanguage === LANGUAGE_US
              ? "headerOption headerOptionHighlighted"
              : "headerOption"
          }
          role="button"
          id="usLanguage"
          data-testid="usLanguage"
          tabIndex={0}
          onKeyPress={() => changeLanguage(LANGUAGE_US)}
        >
          {t("us")}
        </div>
      </div>
    </header>
  );
};

Header.propTypes = {
  changeLocation: PropTypes.func.isRequired,
  changeLanguage: PropTypes.func.isRequired,
  currentLocation: PropTypes.string.isRequired,
  currentLanguage: PropTypes.string.isRequired,
};

const mapStateToProps = (state) => {
  return {
    currentLocation: state.configReducer.currentLocation,
    currentLanguage: state.configReducer.currentLanguage,
  };
};

export default connect(mapStateToProps)(Header);
