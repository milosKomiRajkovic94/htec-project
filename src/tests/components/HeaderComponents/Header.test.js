import React from "react";

import ReactDOM from "react-dom";

import { Provider } from "react-redux";
import { render, cleanup } from "@testing-library/react";
import renderer from "react-test-renderer";
import "@testing-library/jest-dom";
import Header from "../../../components/HeaderComponents/Header";

import configureStore from "../../../store";

const store = configureStore();

afterEach(cleanup);

it("renders without crashing with four required props", () => {
  const div = document.createElement("div");
  ReactDOM.render(
    <Provider store={store}>
      <Header
        changeLanguage={() => {
          console.log("Change language");
        }}
        changeLocation={() => {
          console.log("Change location");
        }}
      />
    </Provider>,
    div
  );
});

it("renders language buttons correctly", () => {
  const { getByTestId } = render(
    <Provider store={store}>
      <Header
        changeLanguage={() => {
          console.log("Change language");
        }}
        changeLocation={() => {
          console.log("Change location");
        }}
      />
    </Provider>
  );
  expect(getByTestId("gbLanguage")).toHaveTextContent("gb");
  expect(getByTestId("usLanguage")).toHaveTextContent("us");
});

it("matches snapshot", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <Header
          changeLanguage={() => {
            console.log("Change language");
          }}
          changeLocation={() => {
            console.log("Change location");
          }}
        />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
