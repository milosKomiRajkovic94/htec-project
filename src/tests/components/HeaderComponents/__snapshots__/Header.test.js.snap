// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<header
  className="header"
>
  <div
    className="headerThreeLeftOptions"
  >
    <div
      className="headerOption headerOptionHighlighted"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      topNews
    </div>
    <div
      className="headerOption"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      categories
    </div>
    <div
      className="headerOption"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      search
    </div>
  </div>
  <div
    className="headerTwoRightOptions"
  >
    <div
      className="headerOption headerOptionHighlighted "
      data-testid="gbLanguage"
      id="gbLanguage"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      gb
    </div>
    <div
      className="headerOption "
      data-testid="usLanguage"
      id="usLanguage"
      onClick={[Function]}
      onKeyPress={[Function]}
      role="button"
      tabIndex={0}
    >
      us
    </div>
  </div>
</header>
`;
